software/apps/audio_module/event_detector/main.c

```diff
@@ -0,0 +1,161 @@
+        0x4032,0x3f68,0x3e9f,0x3dd6,0x3d0c,0x3c43,0x3b7a,0x3ab1,0x39e9,0x3920,0x3858,0x3790,0x36c9,0x3601,0x353b,0x3474,0x33ae,0x32e9,0x3224,0x3160,0x309c,0x2fd9,0x2f16,0x2e54,0x2d93,0x2cd2,0x2c12,0x2b53,0x2a95,0x29d8,0x291b,0x285f,0x27a5,0x26eb,0x2632,0x257a,0x24c3,0x240d,0x2359,0x22a5,0x21f3,0x2141,0x2091,0x1fe2,0x1f35,0x1e88,0x1ddd,0x1d34,0x1c8b,0x1be4,0x1b3f,0x1a9a,0x19f8,0x1956,0x18b7,0x1818,0x177c,0x16e0,0x1647,0x15af,0x1519,0x1484,0x13f1,0x1360,0x12d0,0x1242,0x11b6,0x112c,0x10a4,0x101d,0x0f98,0x0f15,0x0e94,0x0e15,0x0d98,0x0d1d,0x0ca4,0x0c2d,0x0bb8,0x0b44,0x0ad3,0x0a64,0x09f7,0x098c,0x0923,0x08bd,0x0858,0x07f6,0x0796,0x0737,0x06dc,0x0682,0x062b,0x05d5,0x0583,0x0532,0x04e3,0x0497,0x044e,0x0406,0x03c1,0x037e,0x033e,0x0300,0x02c4,0x028a,0x0253,0x021f,0x01ec,0x01bd,0x018f,0x0164,0x013c,0x0115,0x00f2,0x00d0,0x00b2,0x0095,0x007b,0x0064,0x004f,0x003c,0x002c,0x001e,0x0013,0x000b,0x0004,0x0001,0x0000};
+    // windowing and prescaling 
+    for (size_t i = 0; i < BUF_LEN; i++){
+        in[i] = S_MUL(frame[i],win[i]) << 4;
+    }
+
+    if ((cfg = kiss_fftr_alloc(BUF_LEN, 0/*is_inverse_fft*/, NULL, NULL)) == NULL){
+        printf("Not enough memory?\n");
+        return 1;
+    }else{
+        kiss_fftr(cfg, in, out);
+        free(cfg);
+    }
+
+    for (size_t k = 0; k < FRE_LEN; k++){
+        spec[k] = MAG(out[k].r,out[k].i);
+    }
+
+    return 0;
+}
+
+static int printMat(char *filename, kiss_fft_scalar *in, size_t N){
+    FILE *fp;
```

Comment:
So... There is no support currently for storing files. In principal, we could implement a filesystem that ships over to the SD card on the main board, but currently I believe it will always fail on `fopen`.